pkg/apis/core/validation/validation.go

```diff
@@ -1266,6 +1266,11 @@ func validateProjectionSources(projection *core.ProjectedVolumeSource, projectio
 
 			allErrs = append(allErrs, ValidateSignerName(projPath.Child("signerName"), source.PodCertificate.SignerName)...)
 
+			if source.PodCertificate.UserConfig != nil {
+				userConfigErrors := ValidateAnnotations(source.PodCertificate.UserConfig, projPath.Child("userConfig"))
```

Comment:
This validation will need to be customized so that all keys are prefixed with a domain name.